Program.cs

```diff
@@ -46,169 +48,108 @@ public static void Main()
 
         bool gameOver = false;
 
-        int squareSize = windowWidth / gridRows; 
+        int squareSize = windowWidth / gridRows;
         int radius = squareSize / 2;
 
         double startingSpeed = 1.0;
-        double speedFactor = 0.8; 
+        double speedFactor = 0.8;
 
-        int[,] D = { { 0, 1 }, { 1, 0 }, { 0, -1 }, { -1, 0 } };
+        int maximumLength = 11;
 
         Raylib.InitWindow(windowWidth, windowHeight, "Snek");
 
+        Snek snake = new(maximumLength);
+
         // STATE SECTION 
         // Just means all the things that your program has to keep in mind while it's running.
         // (maybe updated each game loop)
-        // Location of snek. 
-        int snakeRow = 10;
-        int snakeCol = 10;
-
-        int topScore = 11;
-        int[] prevLocRow = new int[topScore];
-        int[] prevLocCol = new int[topScore];
-
-        int direction = 0;
-       
         double lastFrameMoved = 0;
         double secondsToMove = startingSpeed;
 
         int score = 0;
 
-         // I need to randomise the location of the apple here so it only does it once per run not every loop        
-        int appleRow = rng.Next(0, gridRows);
-        int appleCol = rng.Next(0, gridCols);
-        Vector2 center = CellToCenter(appleRow, appleCol, squareSize);
+        // I need to randomise the location of the apple here so it only does it once per run not every loop 
+        Apple apple = new();
+
+        apple.SetPosition(gridRows, gridCols);
+
+        Vector2 center = CellToCenter(apple.appleRow, apple.appleCol, squareSize);
 
         // FIGURING OUT THE APPLE OVERLAP 
         for (int i = 0; i < score; i++)
         {
-            if (appleRow == prevLocRow[i] && appleCol == prevLocCol[i])
+            if (apple.appleRow == snake.prevLocRow[i] && apple.appleCol == snake.prevLocCol[i])
```

Comment:
Why would you name the attribute `appleRow` when you can just name it `row`. Then youd have `apple.row`